Visual Studio 2017 csproj format

AnthonySteele opened the pull request:
Use the new VS2017 .csproj syntax. But not yet targeting .Net core. However do want to also target .Net core soon, so we want this new project format.

We have to use the Nunitlite runner. Tests are passing, locally and in appveyor.

Commit:
It builds

Commit:
Use current release version of nlog

Commit:
Added common props

Commit:
Tell appveyor about VS2017

stuart-lang commented on src/NLog.StructuredLogging.Json/NLog.StructuredLogging.Json.csproj:
```diff
@@ -1,73 +1,23 @@
-    <ProjectGuid>{E0357801-F86A-4F85-8246-1E06472C57B2}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <PackageId>NLog.StructuredLogging.Json</PackageId>
```
This can be omitted as it is defaulted to the `.csproj` file name, although if you want it to be explicit that's fine.

stuart-lang commented on src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:
```diff
@@ -1,140 +1,29 @@
-    <ProjectGuid>{EE3EAA37-A6A8-45A3-9E15-31E42FF42959}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <PackageId>NLog.StructuredLogging.Json.Tests</PackageId>
```
This can be omitted.

stuart-lang commented on src/NLog.StructuredLogging.Json.Tests/Properties/AssemblyInfo.cs:
```diff
@@ -1,35 +1,4 @@
 [assembly: ComVisible(false)]
-
-// The following GUID is for the ID of the typelib if this project is exposed to COM
 [assembly: Guid("b482187d-50ac-4c82-971c-e3ee284f5809")]
```
Do we need this? 🤷‍♂️ <- apparently github doesn't do gender neutral shrugs

stuart-lang commented on src/NLog.StructuredLogging.Json/Properties/AssemblyInfo.cs:
```diff
@@ -1,35 +1,4 @@
-// Setting ComVisible to false makes the types in this assembly not visible
-// to COM components.  If you need to access a type in this assembly from
-// COM, set the ComVisible attribute to true on that type.
 [assembly: ComVisible(false)]
```
Is this something we care about keeping?

AnthonySteele replied:
I guess not. So the entire `AssemblyInfo.cs` goes away

stuart-lang reviewed (commented):
Some minor comments, looks good. +68 -281, gotta love it!

Commit:
Simply some more after comments from stuart-lang

martincostello commented on src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:
```diff
@@ -1,140 +1,26 @@
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+    <TreatSpecificWarningsAsErrors />
```
Delete empty property.

martincostello commented on src/NLog.StructuredLogging.Json/NLog.StructuredLogging.Json.csproj:
```diff
@@ -1,73 +1,21 @@
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+    <TreatSpecificWarningsAsErrors />
```
Ditto.

Commit:
remove empty props

Commit:
Test project needs magic guid

Commit:
Try nunitlite

Commit:
Fix some of the failing tests - nlog.config must be included

Commit:
Fix remaining tests
Need this in order to get line numbers on exceptions!

Commit:
Appveyor test script

Commit:
release mode

Commit:
padding the script, since it didn't go to github

Commit:
Unpad

Commit:
Can't guarantee that local date is not offset

Commit:
Fake time should be consistently UTC

Commit:
Looser line length check
the difference is in how lines in the call stack are reported, due to debugging options that aren't that relevant as the client program of this lib will decide.

Commit:
Same issues as before
- one datetime without Kind
- one line length affected by stack trace formats

Commit:
More tests fixed

Commit:
No need for Condition in csproj, there's only one target so far

martincostello commented:
Just a few comments on the AppVeyor build:

  * Doesn't appear to be creating the ```nupkg``` file.
  * Still has the AssemblyInfo patching turned on (warning near the top).
  * The AppVeyor build numbers are ```1.x.x```, should they be ```2.x.x``` now?

Commit:
updates to appveyor file

Commit:
Version is 1.1.x for now

martincostello commented:
I think it's still looking in the wrong place to publish the ```nupkg``` to AppVeyor.

Commit:
Output path is one more up in this solution
as there is the /src/ folder

martincostello commented:
👍

Commit:
Common props
as the prophecy foretold

martincostello commented:
Looks good to go.

stuart-lang commented on src/NLog.StructuredLogging.Json/NLog.StructuredLogging.Json.csproj:
```diff
@@ -1,73 +1,19 @@
-      <HintPath>..\..\packages\NLog.4.4.3\lib\net45\NLog.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <PackageReference Include="Microsoft.NETCore.Platforms" Version="1.1.0" />
```
Don't think this is needed?

stuart-lang commented on src/NLog.StructuredLogging.Json/NLog.StructuredLogging.Json.csproj:
```diff
@@ -1,73 +1,19 @@
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{E0357801-F86A-4F85-8246-1E06472C57B2}</ProjectGuid>
     <OutputType>Library</OutputType>
```
`OutputType` defaults to `Library` fyi, if you want it explicit that's good too.

stuart-lang commented on src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:
```diff
@@ -1,140 +1,37 @@
-      <HintPath>..\..\packages\NUnit.3.6.1\lib\net45\nunit.framework.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
+    <PackageReference Include="Microsoft.NETCore.Platforms" Version="1.1.0" />
```
See other comment

stuart-lang commented on src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:
```diff
@@ -1,140 +1,37 @@
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+  <PropertyGroup>
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Dupe (see common)

stuart-lang commented on src/NLog.StructuredLogging.Json/NLog.StructuredLogging.Json.csproj:
```diff
@@ -1,73 +1,19 @@
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+  <PropertyGroup>
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
```
Dupe

stuart-lang reviewed (commented):
Looks good in general =)

Commit:
Remove some redundant things
